test/blocklib/core/selector.hpp

```diff
@@ -0,0 +1,158 @@
+#ifndef GRAPH_PROTOTYPE_SELECTOR_HPP
+#define GRAPH_PROTOTYPE_SELECTOR_HPP
+
+#include <node.hpp>
+#include <utils.hpp>
+
+namespace gr::blocks::basic {
+using namespace fair::graph;
+
+// optional shortening
+template<typename T, fair::meta::fixed_string description = "", typename... Arguments>
+using A           = Annotated<T, description, Arguments...>;
+
+using SelectorDoc = Doc<R""(
+@brief basic multiplexing class to route arbitrary inputs to outputs
+)"">;
+
+template<typename T>
+struct Selector : node<Selector<T>, SelectorDoc> {
+    // port definitions
+    PortIn<std::uint32_t, Async, Optional> selectOut;
```

Comment:
The `selectOut` and `_selectedSrc` control (if in range), which input is routed to the monitorOut port. The inputs should not be affected regardless if data is being monitored or not.